api/notd/token_manager.py

```diff
@@ -206,6 +209,21 @@ async def update_collection_tokens(self, address: str, shouldForce: bool = False
         await self.update_collection_deferred(address=address, shouldForce=shouldForce)
         await self.update_token_metadatas_deferred(collectionTokenIds=collectionTokenIds, shouldForce=shouldForce)
 
+    async def update_activity_for_collection(self, address: str, date: datetime.datetime):
+        async with self.saver.create_transaction() as connection:
+            retrievedCollectionStats = await self.collectionActivityProcessor.calculate_collection_hourly_activity(registryAddress=address, date=date)
+            await self.saver.create_collection_hourly_activity(connection=connection, address=address, date=retrievedCollectionStats.date, transferCount=retrievedCollectionStats.transferCount, totalVolume=retrievedCollectionStats.totalVolume, minimumValue=retrievedCollectionStats.minimumValue, maximumValue=retrievedCollectionStats.maximumValue, averageValue=retrievedCollectionStats.averageValue)
+
+    async def save_collection_hourly_activity_deferred(self, address: str, date: datetime.datetime, shouldForce: bool=False):
+        await self.tokenQueue.send_message(message=UpdateActivityForCollectionMessageContent(address=address, date=date, shouldForce=shouldForce))
+
+    async def save_collections_hourly_activity_deferred(self, collections: List[Tuple[str,str]]) -> None:
```

Comment:
something is wierd here - why is collections a list of tuples? if the saecond thing is a date this is typed wrong. and more importantly this variable name is quite bad.